codec-memcache/src/main/java/io/netty/handler/codec/memcache/binary/AbstractBinaryMemcacheMessage.java

```diff
@@ -167,10 +167,27 @@ public BinaryMemcacheMessage setOpcode(byte opcode) {
 
     @Override
     public int refCnt() {
+        int extrasCount = 0;
+        int keyCount = 0;
+
+        if (extras == null && key == null) {
+            return 1;
+        }
+
         if (extras != null) {
-            return extras.refCnt();
+            extrasCount = extras.refCnt();
+        }
+        if (key != null) {
+            keyCount = key.refCnt();
         }
-        return 1;
+
+        if (keyCount == extrasCount) {
+            return keyCount;
+        }
+
+        // TODO: not sure this is right at all.
```

Comment:
@daschl I think I would not delegate at all but let AbstractBinaryMemcacheMessage extends AbstractReferenceCounted.